detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportTransformer.java

```diff
@@ -30,17 +30,23 @@
 public class GradleReportTransformer {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
     private final ExternalIdFactory externalIdFactory;
+    private final boolean includeUnresolvedConfigurations;
 
-    public GradleReportTransformer(ExternalIdFactory externalIdFactory) {
+    public GradleReportTransformer(ExternalIdFactory externalIdFactory, boolean includeUnresolvedConfigurations) {
         this.externalIdFactory = externalIdFactory;
+        this.includeUnresolvedConfigurations = includeUnresolvedConfigurations;
     }
 
     public CodeLocation transform(GradleReport gradleReport) {
         MutableDependencyGraph graph = new MutableMapDependencyGraph();
 
         for (GradleConfiguration configuration : gradleReport.getConfigurations()) {
-            logger.trace(String.format("Adding configuration to the graph: %s", configuration.getName()));
-            addConfigurationToGraph(graph, configuration);
+            if (configuration.isUnresolved() && !includeUnresolvedConfigurations) {
```

Comment:
I'd prefer it if we flipped this and maybe had an isResolved instead.
`if (configuration.isResolved || includeUnresolvedConfigurations) { include code } else { log }`